### PR TITLE
emdashii keymap working

### PR DESCRIPTION
**Pull request by emdashii**
## Description

I fixed bugs with the crkbd/keymaps/emdashiiAnimation keymap, and completely rebuild crkbd/keymaps/emdashii keymap, for a corne keyboard with LEDs. emdashiiAnimation is for a corne keyboard with no LEDs. The animation is on the OLED screen. emdashii lights up different keys depending on which layer the keyboard is on. This is really helpful in keyboard use. There are also more layers in emdashii than emdashiiAnimation. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by emdashii** [comment]
I added a bitmap logo to my emdashii keymap.

File: keyboards/crkbd/keymaps/emdashii/oled.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+  char name = ' ';
+    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) ||
+        (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) { keycode = keycode & 0xFF; }
+  if (keycode < 60) {
+    name = code_to_name[keycode];
+  }
+
+  // update keylog
+  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
```
Comment (drashna) [inline keyboards/crkbd/keymaps/emdashii/oled.c:91]: https://docs.qmk.fm/#/squeezing_avr?id=oled-tweaks

This uses ~1.5kb less firmware space.

File: keyboards/crkbd/keymaps/emdashiiAnimation/keymap.c [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+    char name = ' ';
+    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) ||
+        (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) { keycode = keycode & 0xFF; }
+    if (keycode < 60) {
+        name = code_to_name[keycode];
+    }
+
+    // update keylog
+    snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
```
Comment (drashna) [inline keyboards/crkbd/keymaps/emdashiiAnimation/keymap.c:160]: Same as above

File: keyboards/crkbd/keymaps/emdashii/oled.c (hunk 1 above)
Comment (drashna) [reply keyboards/crkbd/keymaps/emdashii/oled.c:91]: Correct. the snprintf function takes up a significant amount of speace. 

As for a keymap that has already done this, not for the corne specifically.  At least, not that I can remember.  But the above should outline that enough. 

```c
  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
           record->event.key.row, record->event.key.col,
           keycode, name);
```
would become something like: 
```c
    oled_write(get_u8_str(record->event.key.row, ' '), false);
    oled_write_P(PSTR("x"), false);
    oled_write(get_u8_str(record->event.key.col, ' '), false);
    oled_write_P(PSTR(", k"), false);
    oled_write(get_u16_str(keycode, ' '), false);
    oled_write_P(PSTR(" : "), false);
    oled_write(name, false);
```

If you want shorting strings (get_u8_str uses 3 characters, get_u16_str uses 5), you can implement a custom implementation.  You can see how the functions are assembled here: 
https://github.com/qmk/qmk_firmware/blob/40d295abf69a79f67628e06599bfdbe50169aa4c/quantum/quantum.c#L519-L573

Comment (emdashii) [reply keyboards/crkbd/keymaps/emdashii/oled.c]: Thanks for the detailed response, I got the code working. I don't know what the best practice is to add this change to the pull request on github. I think I added it, but I'm not sure.

Comment (drashna) [reply keyboards/crkbd/keymaps/emdashii/oled.c]: Welcome!  And it isn't best practice, specifically, but on any AVR processor, very byte you can get back is worth it, especially on a board like the Corne.

File: keyboards/crkbd/keymaps/emdashiiAnimation/config.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (drashna) [inline keyboards/crkbd/keymaps/emdashiiAnimation/config.h:1]: This folder needs to be all lower case.

**Comment by drashna** [comment]
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

**Pull request closed without merging**